test/Microsoft.AspNet.Razor.Test/TestFiles/CodeGenerator/CS/Output/PrefixedAttributeTagHelpers.Reversed.cs

```diff
@@ -0,0 +1,203 @@
+#pragma checksum "PrefixedAttributeTagHelpers.cshtml" "{ff1816ec-aa5e-4d10-87f7-6f4963833460}" "ad6b545da741ad711339a34893d12d2da6f945c6"
+namespace TestOutput
+{
+    using Microsoft.AspNet.Razor.Runtime.TagHelpers;
+    using System;
+    using System.Threading.Tasks;
+
+    public class PrefixedAttributeTagHelpers
+    {
+        #line hidden
+        #pragma warning disable 0414
+        private TagHelperContent __tagHelperStringValueBuffer = null;
+        #pragma warning restore 0414
+        private TagHelperExecutionContext __tagHelperExecutionContext = null;
+        private TagHelperRunner __tagHelperRunner = null;
+        private TagHelperScopeManager __tagHelperScopeManager = new TagHelperScopeManager();
+        private InputTagHelper2 __InputTagHelper2 = null;
+        private InputTagHelper1 __InputTagHelper1 = null;
+        #line hidden
+        public PrefixedAttributeTagHelpers()
+        {
+        }
+
+        #pragma warning disable 1998
+        public override async Task ExecuteAsync()
+        {
+            __tagHelperRunner = __tagHelperRunner ?? new TagHelperRunner();
+            Instrumentation.BeginContext(33, 2, true);
+            WriteLiteral("\r\n");
+            Instrumentation.EndContext();
+#line 3 "PrefixedAttributeTagHelpers.cshtml"
+   
+    var intDictionary = new Dictionary<string, int>
+    {
+        { "three", 3 },
+    };
+    var stringDictionary = new SortedDictionary<string, string>
+    {
+        { "name", "value" },
+    };
+
+#line default
+#line hidden
+
+            Instrumentation.BeginContext(244, 51, true);
+            WriteLiteral("\r\n\r\n<div class=\"randomNonTagHelperAttribute\">\r\n    ");
+            Instrumentation.EndContext();
+            __tagHelperExecutionContext = __tagHelperScopeManager.Begin("input", true, "test", async() => {
+            }
+            , StartTagHelperWritingScope, EndTagHelperWritingScope);
+            __InputTagHelper2 = CreateTagHelper<InputTagHelper2>();
+            __tagHelperExecutionContext.Add(__InputTagHelper2);
+            if (__InputTagHelper2.IntDictionaryProperty == null)
+            {
+                __InputTagHelper2.IntDictionaryProperty = new System.Collections.Generic.Dictionary<string, int>();
+            }
+#line 15 "PrefixedAttributeTagHelpers.cshtml"
+ __InputTagHelper2.IntDictionaryProperty = intDictionary;
```

Comment:
Slight quirk alert: Each tag helper gets their own dictionary when Razor initializes it, which is quite nice when different helpers have different (higher-priority) individual bound properties matching the prefix.  However setting the dictionary explicitly in the Razor source gives every tag helper the same dictionary instance even if the expression is something like `new Dictionary<string, int>()` due to the way we cache expression results.